src/domain/wallets/index.types.d.ts

```diff
@@ -124,14 +124,29 @@ type WalletTransactionHistoryWithPending = {
   readonly transactions: WalletTransaction[]
 }
 
+type WalletName = string & { readonly brand: unique symbol }
+
+// TODO: split with currency?
+type WalletType =
+  typeof import("./index").WalletType[keyof typeof import("./index").WalletType]
+
+type NewWalletInfo = {
+  accountId: AccountId
+  type: WalletType
+}
+
 type Wallet = {
   readonly id: WalletId
+  readonly accountId: AccountId
+  readonly type: WalletType
   readonly onChainAddressIdentifiers: OnChainAddressIdentifier[]
   onChainAddresses(): OnChainAddress[]
 }
 
 interface IWalletsRepository {
-  findById(walletId: WalletId): Promise<Wallet | RepositoryError>
+  persistNew({ accountId, type }: NewWalletInfo): Promise<Wallet | RepositoryError>
+  findById(id: WalletId): Promise<Wallet | RepositoryError>
+  findByAccountId(accountId: AccountId): Promise<Wallet | RepositoryError>
```

Comment:
the structure  is `accounts 1 -> * wallets` so in theory we should not be able to get a wallet by account Id (this method should not be in this repo)

Reply:
we can get back to account from wallet from this PR. `Wallet` has a pointer back to Account. This is not strictly necessary but a welcome optimization because many queries will comes from a wallet (like a transaction being confirmed) and getting the accountId handy will avoid fetching back the Account (currently User) collection.

Reply:
What happens if we have 2 wallets with the same `accountId` value for e.g.? Which `Wallet` gets returned?

Reply:
ohhh. should be `listByAccountId` and return a list of Wallets. you're right. thanks for pointing this out!

Reply:
actually the function was not used. I have removed it.